pkg/plugins/policies/core/matchers/testdata/matchedpolicies/meshgateways/01.golden.yaml

```diff
@@ -39,67 +39,133 @@ FromRules:
-    Subset: []
-  127.0.0.1:8081:
-  - Conf:
-      backends:
-      - file:
-          path: /servicesubset
-        type: File
-    Origin:
-    - creationTime: "0001-01-01T00:00:00Z"
-      mesh: mesh-1
-      modificationTime: "0001-01-01T00:00:00Z"
-      name: gateway
-      type: MeshAccessLog
-    - creationTime: "0001-01-01T00:00:00Z"
-      mesh: mesh-1
-      modificationTime: "0001-01-01T00:00:00Z"
-      name: mesh
-      type: MeshAccessLog
-    - creationTime: "0001-01-01T00:00:00Z"
-      mesh: mesh-1
-      modificationTime: "0001-01-01T00:00:00Z"
-      name: servicesubset
-      type: MeshAccessLog
-    Subset: []
-  127.0.0.1:8082:
-  - Conf:
-      backends:
-      - file:
-          path: /to-gateway
-        type: File
-    Origin:
-    - creationTime: "0001-01-01T00:00:00Z"
-      mesh: mesh-1
-      modificationTime: "0001-01-01T00:00:00Z"
-      name: gateway
-      type: MeshAccessLog
-    - creationTime: "0001-01-01T00:00:00Z"
-      mesh: mesh-1
-      modificationTime: "0001-01-01T00:00:00Z"
-      name: mesh
-      type: MeshAccessLog
-    Subset: []
+  ByListener:
+    127.0.0.1:8080:
+    - Conf:
+        backends:
+        - file:
+            path: /gateway-listener
+          type: File
+      Origin:
+      - creationTime: "0001-01-01T00:00:00Z"
+        mesh: mesh-1
+        modificationTime: "0001-01-01T00:00:00Z"
+        name: gateway
+        type: MeshAccessLog
+      - creationTime: "0001-01-01T00:00:00Z"
+        mesh: mesh-1
+        modificationTime: "0001-01-01T00:00:00Z"
+        name: gatewaylistener
+        type: MeshAccessLog
+      - creationTime: "0001-01-01T00:00:00Z"
+        mesh: mesh-1
+        modificationTime: "0001-01-01T00:00:00Z"
+        name: mesh
+        type: MeshAccessLog
+      Subset: []
+    127.0.0.1:8081:
+    - Conf:
+        backends:
+        - file:
+            path: /servicesubset
+          type: File
+      Origin:
+      - creationTime: "0001-01-01T00:00:00Z"
+        mesh: mesh-1
+        modificationTime: "0001-01-01T00:00:00Z"
+        name: gateway
+        type: MeshAccessLog
+      - creationTime: "0001-01-01T00:00:00Z"
+        mesh: mesh-1
+        modificationTime: "0001-01-01T00:00:00Z"
+        name: mesh
+        type: MeshAccessLog
+      - creationTime: "0001-01-01T00:00:00Z"
+        mesh: mesh-1
+        modificationTime: "0001-01-01T00:00:00Z"
+        name: servicesubset
+        type: MeshAccessLog
+      Subset: []
+    127.0.0.1:8082:
+    - Conf:
+        backends:
+        - file:
+            path: /to-gateway
+          type: File
+      Origin:
+      - creationTime: "0001-01-01T00:00:00Z"
+        mesh: mesh-1
+        modificationTime: "0001-01-01T00:00:00Z"
+        name: gateway
+        type: MeshAccessLog
+      - creationTime: "0001-01-01T00:00:00Z"
+        mesh: mesh-1
+        modificationTime: "0001-01-01T00:00:00Z"
+        name: mesh
+        type: MeshAccessLog
+      Subset: []
+  ByListenerAndHostname:
```

Comment:
Isn't just `ByListener` a `ByListenerAndHostname` with `*` as a hostname? Does having two entries give us anything else?

Reply:
In some sense yes, but the key thing is that it depends on the policy whether targeting hostnames are supported and currently only MeshHTTPRoute supports it. So for MeshHTTPRoute only `ByListenerAndHostname` is set and for other policies only `ByListener`. It makes the distinction a little clearer that for other policies the hostname can _only_ be `*`.